worker/gouging.go

```diff
@@ -156,20 +164,36 @@ func checkFormContractGouging(gs api.GougingSettings, hs rhpv2.HostSettings) err
 	return nil
 }
 
-func uploadPricePerTB(hs rhpv2.HostSettings) types.Currency {
-	sectorPrice := hs.SectorAccessPrice.
-		Add(hs.BaseRPCPrice).
-		Add(hs.UploadBandwidthPrice.Mul64(modules.SectorSize))
-
-	return sectorPrice.Mul64(1 << 40 / modules.SectorSize) // sectors per TiB
+func uploadPricePerTB(hs rhpv2.HostSettings) (types.Currency, bool) {
+	uploadPrice, overflow := hs.UploadBandwidthPrice.Mul64WithOverflow(modules.SectorSize)
+	if overflow {
+		return types.ZeroCurrency, true
+	}
+	sectorPrice, overflow := hs.SectorAccessPrice.AddWithOverflow(hs.BaseRPCPrice)
+	if overflow {
+		return types.ZeroCurrency, true
+	}
+	sectorPrice, overflow = sectorPrice.AddWithOverflow(uploadPrice)
+	if overflow {
+		return types.ZeroCurrency, true
+	}
+	return sectorPrice.Mul64WithOverflow(1 << 40 / modules.SectorSize) // sectors per TiB
 }
 
-func downloadPricePerTB(hs rhpv2.HostSettings) types.Currency {
-	sectorPrice := hs.SectorAccessPrice.
-		Add(hs.BaseRPCPrice).
-		Add(hs.DownloadBandwidthPrice.Mul64(modules.SectorSize))
-
-	return sectorPrice.Mul64(1 << 40 / modules.SectorSize) // sectors per TiB
+func downloadPricePerTB(hs rhpv2.HostSettings) (types.Currency, bool) {
+	downloadPrice, overflow := hs.DownloadBandwidthPrice.Mul64WithOverflow(modules.SectorSize)
```

Comment:
there's zero chance of keeping an `overflow` bool in the `Currency` type itself I guess 
```go
sectorPrice, overflow := hs.SectorAccessPrice.
		SafeAdd(hs.BaseRPCPrice).
		SafeAdd(hs.DownloadBandwidthPrice.SafeMul64(modules.SectorSize)).
		Finalize()
```